src/main/java/org/jenkinsci/plugins/workflow/multibranch/ReadTrustedStep.java

```diff
@@ -248,7 +248,7 @@ private static boolean isDescendant(FilePath child, FilePath parent) throws IOEx
             if (child.isRemote() || parent.isRemote()) {
                 throw new IllegalStateException();
             }
-            return new File(child.getRemote()).getCanonicalFile().toPath().startsWith(parent.absolutize().getRemote());
+            return new File(child.getRemote()).getCanonicalFile().toPath().startsWith(new File(parent.getRemote()).getCanonicalPath());
```

Comment:
The only intended difference here is that if `JENKINS_HOME` is a symlink, the code should no longer throw an exception in non-dangerous scenarios.